torch/csrc/autograd/VariableTypeUtils.h

```diff
@@ -134,7 +134,7 @@ template<typename... Args> inline variable_list flatten_tensor_args(Args&&... ar
 }
 
 // See NOTE [ Autograd View Variables ] for details.
-inline Tensor as_view(const Tensor & base, Tensor tensor, bool is_differentiable,
+inline Tensor as_view(const Tensor & base, Tensor& tensor, bool is_differentiable,
```

Comment:
I'm actually pretty confused; `tensor` looks completely dead in the body here. What's going on?!

Reply:
it was std::moved into make_variable_differentiable_view below.

Reply:
OK, somehow, I didn't see the site when I was ctrl-f'ing.

Are you sure this new change is right? If the caller to `as_view` expected to retain an owning reference to tensor, the move below is wrong, and taking it by value here was actually the correct thing to do. Hard to say without looking at the call sites. If you do intend to move out of tensor, `Tensor&&` is a more accurate type signature.

Reply:
@ezyang Hmmm I think the move is not necessary for `tensor` so I just removed it and make them all pass by reference for `tensor`. Local test_autograd run seem to pass for me. Let me know if this looks better now. Thanks!

Reply:
OK, so since you're not moving out, now you can make this a const reference?

Reply:
> If the caller to as_view expected to retain an owning reference to tensor

I don't think this code ended up retaining the passed tensor before or after this change. Did I miss something?